FileManagerWpf/Services/FileManager.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using FileManagerWpf.Model;
+using FileManagerWpf.ViewModel;
+
+namespace FileManagerWpf.Services
+{
+    public class FileManager
+    {
+
+        private readonly FileSystemWatcher _watcher;
+        private readonly Action _onChanges;
+
+        public FileManager(Action onChanges)
+        {            
+            _watcher = new FileSystemWatcher();
+            _watcher.Filter = "*.*";
+            _watcher.NotifyFilter = NotifyFilters.Attributes
+                                    | NotifyFilters.CreationTime
+                                    | NotifyFilters.DirectoryName
+                                    | NotifyFilters.FileName
+                                    | NotifyFilters.LastAccess
+                                    | NotifyFilters.LastWrite
+                                    | NotifyFilters.Security
+                                    | NotifyFilters.Size;
+
+            _watcher.Changed += OnChanges;
+            _watcher.Created += OnChanges;
+            _watcher.Deleted += OnChanges;
+            _watcher.Renamed += OnChanges;
+            _onChanges = onChanges;          
+        }      
+
+        public IEnumerable<DriveItem> GetDrives()
+        {
+            foreach (var drive in DriveInfo.GetDrives())
+            {
+                yield return new DriveItem
+                {
+                    Path = drive.RootDirectory.FullName,
+                    Name = drive.RootDirectory.Name,
+                    Size = drive.TotalSize,
+                    TotalFreeSpace = drive.TotalFreeSpace,
+                    UsedSpace = drive.TotalSize - drive.TotalFreeSpace
+                };
+            }
+        }
+        public IEnumerable<FolderItem> GetSubDirectories(TabViewModel tabViewModel, string path)
+        {
+            if (!Directory.Exists(path)) throw new DirectoryNotFoundException();
+
+            foreach (var dir in Directory.GetDirectories(path))
```

Comment:
```suggestion
            foreach (var dir in Directory.EnumerateDirectories(path))
```
Данный метод не читает сразу всё содержимое в массив, а получает по одному элементу за итерацию. Работает быстро, не жрёт память.